### ISSUE TITLE
Pack relative paths

### ISSUE THREAD

**Issue by in4lio** [gnuarmeclipse/plug-ins#64]
Please add a ${variable} that will point to the folder of selected CMSIS Pack for the current project.
Something like ${ProjDirPath}.

**Comment by ilg-ul** [comment]
ok, to remain consistent with the other definitions, the first suggestion for the variable name is `packs_path`.

**Comment by in4lio** [comment]
Yes, `packs_path` will point to `Packages` folder.

It would be nice to have variables for all installed devices, like Keil 5 has `$$Device:NAME$` variables, e.g. `$$Device:STM32F103$`points to `c:\Keil5\ARM\Pack\STM\STM32F1xx\2.0.1\`. So we will be able to use `${packs_path}/${STM32F103}`.

Also `${packs_path}/${selected_device}` for the current project, could be very useful, I think)

**Comment by in4lio** [comment]
A small correction, `$$Device:NAME$` variables in Keil also depend on the selected package.

**Comment by ilg-ul** [comment]
I don't think that more package specific variables are realistic, there are too many.

anyway, package management should be automatic, a wizard should create all paths relative to `${packs_path}`, so there should be no need to manually manage so many paths.

I am planning such a new wizard, but it'll take a while to implement it.

**Comment by in4lio** [comment]
This way also seems to be quite reasonable.
